src/pybind/mgr/cephadm/agent.py

```diff
@@ -205,10 +205,7 @@ def handle_metadata(self, data: Dict[str, Any]) -> None:
             error_daemons_old = set([dd.name() for dd in self.mgr.cache.get_error_daemons()])
             daemon_count_old = len(self.mgr.cache.get_daemons_by_host(host))
 
-            agents_down = []
-            for h in self.mgr.cache.get_hosts():
-                if self.mgr.agent_helpers._check_agent(h):
-                    agents_down.append(h)
+            agents_down = self.mgr.agent_helpers._check_agents()
```

Comment:
why do we need to do that in handle_metadata? WDYT of running check_agents *exclusively* in `serve()`? We then wouldn't need any locking at all here.